### DISCUSSION TITLE
Welcome to aws-saas-factory-eks-reference-architecture Discussions!

### DISCUSSION THREAD

**Discussion by tobuck-aws** [aws-samples/aws-saas-factory-eks-reference-architecture#19] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rayjencode** [comment]
Hi my name is Rayjens, a developer and I'm learning the SaaS space for my future multi-tenant project. I just have some questions since I'm new on cloud computing keys and terms. I got really interested after I watched the video presentation of Gunnar Grosch and Ranjith Raman and really thanks to this architecture you guys been created. :) 

 Q 1: About the registration service, I see the source code are base on java and Im not famillar with it. Do I also use Nodejs to create the registration API? 

Q 2: My applications is using mongoDB, do I need to migrate it using the dynamoDB? 

Q 3: The typical e-commerce application has Add products, Create Orders etc. Do I need to microservice this part? 

Q 4: Im using github as my respository, But I didnt see in the video how the repo has been hook from github.

**Reply by oliwerhelsen** [reply]
Hi Rayjen,

I'm not part of the core team here, but I think the answer is Yes to all your questions.

Q1: There is no problem with porting this using JS or TS instead of JAVA.
Q2: I don't think there is a direct link to migrate to DynamoDB. There is a lot that differentiates between those two. But as a start, you can use Mongo as a docker image and then create new services with Dynamo.
Q3: No you don't. I believe that you can go the way you want to. But microservices and lambdas are a nice solution. 
Q4: This project use Codecommit. You can change this but there needs to be some work. I think the Vnext branch is easier to modify if you're unfamiliar with cloud formation. 

Hope this help!
Best regards
Oliwer H.

**Comment by ranjith-aws** [comment]
HI @rayjencode, Thanks for reaching out. I agree with @oliwerhelsen 's answers. For Q1, we do have a workshop where we use NodeJs/Typescript for all the control plane and application services. This is link, https://github.com/aws-samples/aws-saas-factory-eks-saas-workshop. Feel free to check it out and let us know if any other questions.

Also, we are working on a vNext branch in this repo, which is undergoing testing and we are expecting to launch end of September. But most of the code there is stable too, so please check that out as well.
